modern/src/common/components/LocalizationProvider.js

```diff
@@ -178,7 +178,7 @@ export const useLocalization = () => useContext(LocalizationContext);
 export const useTranslation = () => {
   const context = useContext(LocalizationContext);
   const { data } = context.languages[context.language];
-  return useMemo(() => (key) => data[key], [data]);
+  return useMemo(() => (key) => (data[key] ? data[key] : context.languages.en.data[key]), [data]);
```

Comment:
Why do you need the fallback here?

Reply:
So that the frontend doesn't fail when run in another language.

It was more relevant when the translations were being concat-ed, but even now, unless I add translations for every language for "Upload geofence" for example, then nothing will be displayed.

The above change means that the English option will always be displayed, even if there is no local translation.

Reply:
I guess it makes sense, but you should do the same thing as for selected language. Pass it as a variable.

Reply:
Please could you elaborate? I don't understand what you mean..

Reply:
It needs to be passed to `useMemo`.

Reply:
I still don't understand what you mean here. Maybe I should just remove it - it only affects one non-breaking translation

Reply:
Do you see how the selected language is used? You have to do absolutely the same thing with english.

Reply:
I still don't understand what you mean tbh, so I've just removed it. Its only relevant for the "Upload" tooltip which can just remain blank for non-English languages